M/MPItrampoline/build_tarballs.jl

```diff
@@ -0,0 +1,61 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "MPItrampoline"
+version = v"1.0.0"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/eschnett/MPItrampoline/archive/c79516ce4a015ea2e3cf6f417b826f80660a07af.tar.gz",
+                  "a07b0d898f0d39d5ac64baf2116d1fb5d49fa8d6f8ef20f1cc456db718408184"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd MPItrampoline-*
+mkdir build
+cd build
+cmake \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_FIND_ROOT_PATH=$prefix \
+    -DCMAKE_INSTALL_PREFIX=$prefix \
+    -DBUILD_SHARED_LIBS=ON \
+    ..
+cmake --build . --config RelWithDebInfo --parallel $nproc
+cmake --build . --config RelWithDebInfo --parallel $nproc --target install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+# Apple: Dynamically loaded libraries (`dlopen`) are not handled correctly.
+# Windows: Does not have `dlopen`.
+# musl: Does not define `RTLD_DEEPBIND` for `dlopen`.
+platforms = filter(p -> !(Sys.isapple(p) || Sys.iswindows(p) || libc(p) == "musl"), platforms)
+platforms = expand_gfortran_versions(platforms)
+# libgfortran3 does not support `!GCC$ ATTRIBUTES NO_ARG_CHECK`. (We
+# could in principle build without Fortran support there.)
+platforms = filter(p -> libgfortran_version(p) ≠ v"3", platforms)
+
+# TODO: on 32-bit systems, Fortran `MPI_Status` has the wrong size
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("mpicc", :mpicc),
+    ExecutableProduct("mpicxx", :mpicxx),
+    ExecutableProduct("mpifc", :mpifc),
+    ExecutableProduct("mpifort", :mpifort),
+    ExecutableProduct("mpiexec", :mpiexec),
+
+    LibraryProduct("libmpitrampoline", :libmpitrampoline),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
```

Comment:
Can we require Julia v1.6, just to be future-proof?
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
```

Reply:
Sure! What does this do?

Reply:
The generated JLL package will require Julia v1.6.  The main advantage of Julia v1.6 is that it doesn't freak out if there are unknown platforms in `Artifacts.toml`, which lets us add more easily new platforms in the future.  For example, `aarch64-apple-darwin` (Apple Silicon) isn't known to Julia v1.5-, and it doesn't like this if this platform shows up in `Artifacts.toml`

Reply:
Ok. Done.